src/main/kotlin/org/rust/ide/refactoring/changeSignature/RsChangeSignatureHandler.kt

```diff
@@ -0,0 +1,342 @@
+    if (function.returnType != config.returnType) {
+        function.retType?.delete()
+        if (config.returnType !is TyUnit) {
+            val ret = factory.createRetType(config.returnType.renderInsertionSafe())
+            function.addAfter(ret, function.valueParameterList)
+        }
+    }
+}
+
+private fun changeParameters(
+    factory: RsPsiFactory,
+    function: RsFunction,
+    config: RsFunctionSignatureConfig,
+    parameterOps: List<ParameterOperation>
+) {
+    val parameters = function.valueParameterList ?: return
+    val parameterList = parameters.valueParameterList.toList()
+    val originalParameters = parameters.copy() as RsValueParameterList
+
+    fun createType(parameter: Parameter): RsTypeReference =
+        factory.tryCreateType(parameter.type.renderInsertionSafe()) ?: factory.createType("()")
+
+    cycle@ for ((index, data) in config.parameters.zip(parameterOps).withIndex()) {
+        val (parameter, op) = data
+        when (op) {
+            is ParameterOperation.Add -> {
+                deleteItem(parameterList, index)
+
+                val typeReference = createType(parameter)
+                val newParameter = factory.createValueParameter(parameter.patText, typeReference, reference = false)
+                val anchor = findAnchorToInsertItem(parameters, index)
+                insertItemWithComma(factory, newParameter, parameters, anchor)
+            }
+            is ParameterOperation.Move -> {
+                deleteItem(parameterList, index)
+
+                val originalParameter = originalParameters.valueParameterList.getOrNull(op.originalIndex)
+                    ?: continue@cycle
+                val anchor = findAnchorToInsertItem(parameters, index)
+                insertItemWithComma(factory, originalParameter, parameters, anchor)
+            }
+        }
+
+        when (op) {
+            is ParameterOperation.Move, is ParameterOperation.Keep -> {
+                val currentParameter = parameters.valueParameterList[index]
+                if (parameter.pat.text != currentParameter.pat?.text) {
+                    if (parameter.pat is RsPatIdent && currentParameter.pat is RsPatIdent) {
+                        renameParameter(function.project, currentParameter, parameter.patText)
+                    } else {
+                        currentParameter.pat?.replace(parameter.pat)
+                    }
+                }
+                if (parameter.type != currentParameter.typeReference?.type) {
+                    currentParameter.typeReference?.replace(createType(parameter))
+                }
+            }
+        }
+    }
+
+    // remove remaining parameters
+    for (index in parameterOps.size until parameterList.size) {
+        deleteItem(parameterList, index)
+    }
+}
+
+private fun changeArguments(factory: RsPsiFactory, usage: RsElement, parameterOps: List<ParameterOperation>) {
```

Comment:
I wanted to avoid that because I thought that it would be difficult to reconstruct the original list (its whitespace, comments etc.). In other intentions/quick fixes/refactoring I was usually creating things from scratch and @Undin hinted to me that changing existing things should be generally a bit more stable.